shadowsocks-csharp/Controller/Service/TCPRelay.cs

```diff
@@ -566,12 +566,28 @@ private void StartConnect()
             try
             {
                 CreateRemote();
-
+                Server server=new Server(_server);

```

Comment:
MemberwiseClone could be another option. (No worries, no need to change current code. It's fine.)